BEIMA.Backend/Models/Requests.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BEIMA.Backend.Models
+{
+    /// <summary>
+    /// Object representation of the data object in an add device request
+    /// </summary>
+    public class AddDeviceRequest
+    {
+        public string DeviceTag { get; set; }
+        public string DeviceTypeId { get; set; }
+        public string Manufacturer { get; set; }
+        public string ModelNum { get; set; }
+        public string SerialNum { get; set; }
+        public int YearManufactured { get; set; }
+        public string Notes { get; set; }
+        public Location Location { get; set; }
+        public Dictionary<string, string> Fields { get; set; } = new Dictionary<string, string>();
+    }
+
+    /// <summary>
+    /// Object representation of the data object in an updated device request
+    /// </summary>
+    public class UpdateDeviceRequest
+    {
+        public string DeviceTag { get; set; }
+        public string DeviceTypeId { get; set; }
+        public string Manufacturer { get; set; }
+        public string ModelNum { get; set; }
+        public string SerialNum { get; set; }
+        public int YearManufactured { get; set; }
+        public string Notes { get; set; }
+        public Location Location { get; set; }
+        public Dictionary<string, string> Fields { get; set; } = new Dictionary<string, string>();
+        public List<string> DeletedFiles { get; set; } = new List<string>();
```

Comment:
Do we need AddFIles as well? Do we need an object for an updated image?

Reply:
No as those will exist outside of the data object that is being sent up. In a mulitpart form request, added files and photos will be found in req.Files. The difference is the name property attached to each individual file. Its a little weird, but that is a quirk of the request. The Request.cs objects will map to req.Form['data'] and will be how we are able to send the whole device json alongside the files.